vault/dynamic_system_view.go

```diff
@@ -27,19 +27,27 @@ type dynamicSystemView struct {
 	mountEntry *MountEntry
 }
 
-type extendedSystemView struct {
+type extendedSystemView interface {
```

Comment:
There is already a `logical.ExtendedSystemView` interface.

Reply:
I've synced these a bit and added a comment.